client_linux.go

```diff
@@ -148,6 +150,53 @@ func (c *client) linkMode(flags netlink.HeaderFlags, ifi Interface) ([]*LinkMode
 	return parseLinkModes(msgs)
 }
 
+// UpdateLinkMode updates the given Interface with the non-nil link mode properties in
+// the LinkModeUpdate.
+func (c *client) UpdateLinkMode(ifi Interface, lmu *LinkModeUpdate) error {
+	if lmu.Advertise != nil && lmu.Advertise.Sign() < 0 {
+		return errors.New("ethtool: can't update link mode, Advertise is invalid")
+	}
+	_, err := c.get(
+		unix.ETHTOOL_A_LINKMODES_HEADER,
+		unix.ETHTOOL_MSG_LINKMODES_SET,
+		netlink.Acknowledge,
+		ifi,
+		lmu.encode,
+	)
+	return err
+}
+
+// encode packs LinkModeUpdate data into the appropriate netlink attributes for the
+// encoder.
+func (lmu *LinkModeUpdate) encode(ae *netlink.AttributeEncoder) {
+	if lmu.SpeedMegabits != nil {
+		ae.Uint32(unix.ETHTOOL_A_LINKMODES_SPEED, uint32(*lmu.SpeedMegabits))
+	}
+	if lmu.Duplex != nil {
+		ae.Uint8(unix.ETHTOOL_A_LINKMODES_DUPLEX, uint8(*lmu.Duplex))
+	}
+	if lmu.Autoneg != nil {
+		ae.Uint8(unix.ETHTOOL_A_LINKMODES_AUTONEG, uint8(*lmu.Autoneg))
+	}
+	if lmu.Advertise != nil {
+		ae.Nested(unix.ETHTOOL_A_LINKMODES_OURS, func(nae *netlink.AttributeEncoder) error {
+			nae.Flag(unix.ETHTOOL_A_BITSET_NOMASK, true)
+			bitlen := lmu.Advertise.BitLen()
+			nae.Uint32(unix.ETHTOOL_A_BITSET_SIZE, uint32(bitlen))
+			b := make([]byte, ((bitlen+31)/32)*4)
+			b = lmu.Advertise.FillBytes(b)
+			if binary.ByteOrder(native.Endian) == binary.LittleEndian {
+				// FillBytes is big endian, reverse bytes for host order
+				for i, j := 0, len(b)-1; i < j; i, j = i+1, j-1 {
+					b[i], b[j] = b[j], b[i]
+				}
```

Comment:
You can use the slices package now.

```suggestion
				slices.Reverse(b)
```